apps/channel/components/collaborators/new_collaborator_view.coffee

```diff
@@ -0,0 +1,82 @@
+Backbone = require "backbone"
+_ = require 'underscore'
+sd = require("sharify").data
+mediator = require '../../../../lib/mediator.coffee'
+analytics = require '../../../../lib/analytics.coffee'
+Collaborators = require '../../../../collections/collaborators.coffee'
+
+newCollaboratorTemplate = -> require('./templates/new.jade') arguments...
+inviteTemplate = -> require('./templates/invite.jade') arguments...
+resultsTemplate = -> require('./templates/results.jade') arguments...
+
+module.exports = class NewCollaboratorView extends Backbone.View
+
+  events:
+    'keyup .new-collaborator__search' : 'searchCollaborators'
+    'click .js-collaborator-result' : 'addCollaborator'
+    'click .js-collaborator-invite' : 'inviteCollaborator'
+
+  initialize: ({ @channel, @collaborators} )->
+    @render()
+
+  addCollaborator: (e) ->
+    @collaborators._add @results.get $(e.currentTarget).data 'id'
+
+  inviteCollaborator: (e) ->
+    @collaborators._invite $(e.currentTarget).data 'email'
+
+  isEmail: (query)->
+    is_email = new RegExp("[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?")
```

Comment:
Probably want to extract this into a single utility function and use it here rather than it being a method on the view.